vnext/local-cli/config/dependencyConfig.js

```diff
@@ -1,102 +1,117 @@
-const fs = require('fs');
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * @format
+ */
+// @ts-check
+
 const path = require('path');
-const glob = require('glob');
-const xmldoc = require('xmldoc');
 
-function dependencyConfigWindows(folder, userConfig = {}) {
-  const sourceDir = userConfig.sourceDir || findWindowsAppFolder(folder);
+const configUtils = require('./configUtils.js');
+
+/*
+
+Schema for dependencies:
+
+{
+  folder: '', // absolute path to the module root, ex: c:\path\to\node_modules\module-name
+  sourceDir: '', // relative path to the windows implementation folder, ex: windows
+  solutionFile: '', // relative path to the solution, ex: ModuleName.sln
+  projects: [ // array of projects to be built that include modules
+    {
+      projectFile: '', // relative path to the project, ex: ProjectName\ProjectName.vcxproj
+      projectName: '', // name of the project
+      projectLang: '', // language of the project, cpp or cs
+      projectGuid: '', // project identifier
```

Comment:
Is this any guid that the package wants to generate, or does it have to match a guid used by ProjectName.vcxproj?  And if it has to match, can we just extract it from the project file?

Reply:
We do extract it, but the process is decoupled. react-native config does all the extracting and then serves up the data for autolinking (and others?). But I guess for projects where they create their own react-native.config file, we could add some logic to extract it again in case they forget it.

Reply:
Fixed.